frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx

```diff
@@ -342,18 +249,95 @@ export const PrometheusMultilineUtilizationItem = withDashboardResources<
   },
 );
 
-export const UtilizationCard = () => {
+const UtilizationCardNodeFilter: React.FC<UtilizationCardNodeFilterProps> = ({
+  machineConfigPools,
+  onNodeSelect,
+  selectedNodes,
+}) => {
   const { t } = useTranslation();
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+  const sortedMCPs = machineConfigPools.sort((a, b) => {
+    const order = ['worker', 'master'];
+    const indexA = order.indexOf(a.metadata.name);
+    const indexB = order.indexOf(b.metadata.name);
+    if (indexA === -1 && indexB === -1) {
+      return indexA - indexB;
+    }
+    if (indexA === -1) {
+      return 1;
+    }
+    if (indexB === -1) {
+      return -1;
+    }
+    return indexA - indexB;
+  });
+  const onToggle = (open: boolean): void => setIsOpen(open);
 
+  return (
+    <Select
+      variant={SelectVariant.checkbox}
+      aria-label={t('public~Filter by Node type')}
+      onToggle={onToggle}
+      onSelect={onNodeSelect}
+      selections={selectedNodes}
+      isOpen={isOpen}
+      placeholderText={t('public~Filter by Node type')}
+    >
+      {sortedMCPs.map((mcp) => (
+        <SelectOption key={mcp.metadata.name} value={mcp.metadata.name} />
+      ))}
+    </Select>
+  );
+};
+
+export const UtilizationCard = () => {
+  const { t } = useTranslation();
   const [timestamps, setTimestamps] = React.useState<Date[]>();
   const [duration, setDuration] = useMetricDuration(t);
+  const [machineConfigPools, machineConfigPoolsLoaded] = useK8sWatchResource<
+    MachineConfigPoolKind[]
+  >({
+    isList: true,
+    kind: referenceForModel(MachineConfigPoolModel),
+  });
+  // TODO: add `useUserSettings` to get default selected
+  const [selectedNodes, setSelectedNodes] = React.useState<string[]>([]);
+  // TODO: add `useUserSettingsCompatibility` to store selectedNodes
+  const onNodeSelect = (event: React.MouseEvent, selection: string) => {
+    if (selectedNodes.includes(selection)) {
+      setSelectedNodes(selectedNodes.filter((item) => item !== selection));
+    } else {
+      setSelectedNodes([...selectedNodes, selection]);
+    }
+  };
+  // if no filter is applied, show all nodes using regex
+  const nodeType = _.isEmpty(selectedNodes) ? '.+' : selectedNodes.join('|');
+  const consumerQueries = getTop25ConsumerQueries(nodeType);
```

Comment:
We should be consistent in our use of `useMemo`

```suggestion
  const consumerQueries = React.useMemo(() => getTop25ConsumerQueries(nodeType), [nodeType]);
```

Reply:
Added.